src/_guides/language/extension-methods.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Extension methods
+description: Learn how to add to existing APIs.
+---
+<?code-excerpt replace="/ *\/\/\s+ignore_for_file:[^\n]+\n//g; /([A-Z]\w*)\d\b/$1/g; /\b(main)\d\b/$1/g"?>
+
+Extension methods, introduced in Dart 2.7,
+are a way to add functionality to existing libraries.
+You might not ever need to _use_ extension methods, much less _create_ them.
+But if you generate code or find yourself writing lots of wrappers,
+consider using extension methods to reduce
+the amount of boilerplate code that you need to write.
+
+You might use extension methods without even knowing it.
+For example, when you use code completion in an IDE,
+it suggests extension methods alongside regular methods.
+
+{% comment %}
+[PENDING: embed video here.]
+Watch this video for an introduction to extension methods.
+{% endcomment %}
+
+## Overview
+
+When you’re using someone else’s API or
+when you implement a library that’s widely used,
+it’s often impractical or impossible to change the API.
+But you might still want to add some functionality.
+
+For example, consider the following code that parses a string into an integer:
+
+```dart
+int.parse('42')
+```
+
+It might be nice — shorter and easier to use with tools — to
+have that functionality be in the string instead:
+
+```dart
+'42'.toInt()
+```
+
+To enable that code,
+you can import a library that contains an extension of the String class:
+
+```dart
+import string_apis.dart';
+...
+print('42'.toInt());
+```
+
+Extensions can define not just methods and getters,
+but also other members such as setters and operators.
+Also, extensions have names, which can be helpful if an API conflict arises.
+Here's how you might implement the extension method `toInt()`,
+using an extension (named `ParseNumbers`) that operates on strings:
+
+```dart
+extension ParseNumbers on String {
+  int toInt() {
+    return int.parse(this);
+  }
+}
+```
+
+The next section describes how to _use_ extension methods.
+After that are sections about _implementing_ extension methods.
+
+
+## Using extension methods
+
+Like all Dart code, extension methods are in libraries.
```

Comment:
General comment, outside of this feature: I wonder how often developers deal with libraries? Is it not much more common to deal with (import etc.) packages these days?

Reply:
I'm not sure what you mean... You deal with packages in the pubspec, but in Dart code you import and deal with library files (which are often inside of packages), no? And when I go to the API reference, I see libraries (though the imports aren't usually in the API docs, sadly).